packages/react-router/src/utils.ts

```diff
@@ -131,15 +131,6 @@ export function useIntersectionObserver<T extends Element>(
  */
 export function useForwardedRef<T>(ref?: React.ForwardedRef<T>) {
   const innerRef = React.useRef<T>(null)
-
-  React.useEffect(() => {
-    if (!ref) return
-    if (typeof ref === 'function') {
-      ref(innerRef.current)
-    } else {
-      ref.current = innerRef.current
-    }
-  })
-
+  React.useImperativeHandle(ref, () => innerRef.current!, [])
```

Comment:
```suggestion
  React.useImperativeHandle(ref, () => innerRef.current!)
```

Maybe we should leave the dep array out, if we want to be sure this reproduces the same behavior as the `useEffect` it replaces

Reply:
I tested with an empty deps array and it seemed fine: https://stackblitz.com/edit/vitejs-vite-usxaksas?file=src%2FApp.tsx&terminal=dev

The cost of removing the deps array and re-executing this on every render is probably low, but in this case (a regular DOM ref, and not an object with a closure scope that matters), using `[]` is fine because the value is static. At least that's my understanding.

Reply:
Are you satisfied with the PR? If so, I'm happy to merge.

Reply:
Yeah let's go, i hope I'm not breaking anything, but I'm pretty confident this is correct.